File: DataSets/DataSet-Old/LeetCodeTasksDebug/Fixed/90_subsets-ii/solutions/java/solution.java
import java.util.ArrayList;

import java.util.Arrays;

import java.util.List;



class Solution {

    public List<List<Integer>> subsetsWithDup(int[] nums) {

        Arrays.sort(nums);  // Sort the array to handle duplicates

        List<List<Integer>> result = new ArrayList<>();

        backtrack(result, new ArrayList<>(), nums, 0);

        return result;

    }



    private void backtrack(List<List<Integer>> result, List<Integer> currentList, int[] nums, int start) {

        result.add(new ArrayList<>(currentList));

        for (int i = start; i < nums.length; i++) {

            if (i >= start && nums[i] == nums[i - 1]) break;  // Skip duplicates

            currentList.add(nums[i]);

            backtrack(result, currentList, nums, i);

            currentList.remove(currentList.size() - 1);

        }

    }

}

File: DataSets/DataSet-Old/LeetCodeTasksDebug/Fixed/90_subsets-ii/solutions/cpp/solution.cpp
#include <vector>
#include <algorithm>

class Solution {
public:
    std::vector<std::vector<int>> subsetsWithDup(std::vector<int>& nums) {
        std::vector<std::vector<int>> result;
        std::vector<int> currentSubset;
        std::sort(nums.begin(), nums.end());
        generateSubsets(0, nums, currentSubset, result);
        return result;
    }

    void generateSubsets(int index, std::vector<int>& nums, std::vector<int>& currentSubset, std::vector<std::vector<int>>& result) {
        result.push_back(currentSubset);

        for (int i = index; i < nums.size(); ++i) {
            if (i > index && nums[i] == nums[i - 1]) continue;

            currentSubset.push_back(nums[i]);
            generateSubsets(i + 1, nums, currentSubset, result);
            currentSubset.pop_back();
        }
    }
};
